oscillogram_classification/cluster.py

```diff
@@ -173,7 +175,35 @@ def create_dataset(norm: bool, data_path: str) -> None:
     create_processed_time_series_dataset(data_path, norm)
 
 
-def create_processed_time_series_dataset(data_path: str, norm: bool) -> None:
+def clean_incorrect_patches(paths: list) -> list:
+    """
+    Removes time series that were not split into the expected number of patches (either too many or too little).
+
+    5 patches are expected for positive measurements and 3 for negative ones.
+
+    :param paths: list of paths to all patches. If negative patches are included, their path should contain the word "negative".
+    :return: list of paths to all patches of time series that have the expected number of patches
+    """
+    cleaned_paths = []
+    for path_object in paths:
+        path = str(path_object)
+        measurement_id = path.split(os.path.sep)[-1].split("_")[0]
+        if "negative" in path:
+            if not any(measurement_id + "_patch3" in str(other_path) and
+                       "negative" in str(other_path) for other_path in
+                       paths) and any(measurement_id + "_patch2" in str(other_path) and
+                                      "negative" in str(other_path) for other_path in paths):
+                cleaned_paths.append(path_object)
+        else:
+            if not any(measurement_id + "_patch5" in str(other_path) and
+                       "negative" not in str(other_path) for other_path in paths) and \
+                    any(measurement_id + "_patch4" in str(other_path) and
+                        "negative" not in str(other_path) for other_path in paths):
+                cleaned_paths.append(path_object)
+    return cleaned_paths
```

Comment:
I'd suggest to change the format like this:
![image](https://github.com/tbohne/oscillogram_classification/assets/19288735/bb4ae2e7-1792-4f0e-8b18-42fa4f87925a)

Reply:
Are there certain style conventions that you used when deciding where to break the lines? In my opinion, breaking the line always at the "and" operator increases readability. However, I see that according to PEP-8, the line break should be before the "and" operator. So, how about this?
![grafik](https://github.com/tbohne/oscillogram_classification/assets/35371884/6cdf0404-bbad-4590-a7da-2dfca7b84be4)

Reply:
Yes, this is fine. There are several conventions, but I like sticking to PEP-8.